src/Compiler/Driver/GraphChecking/FileContentMapping.fs

```diff
@@ -320,14 +339,20 @@ let (|NameofExpr|_|) (e: SynExpr) =
     match e with
     | SynExpr.App(flag = ExprAtomicFlag.NonAtomic; isInfix = false; funcExpr = SynExpr.Ident NameofIdent; argExpr = moduleNameExpr) ->
         match stripParen moduleNameExpr with
-        | SynExpr.Ident moduleNameIdent -> Some moduleNameIdent
+        | SynExpr.Ident moduleNameIdent -> Some(NameofResult.SingleIdent moduleNameIdent)
+        | SynExpr.LongIdent(longDotId = longIdent) ->
```

Comment:
@nojaf I think this may require more cases that can be parsed here and where a module name is allowed. For example, consider this:
```fsharp
nameof GenericType<Module.TypeInsideModule>
```
Maybe I'm missing something, but I thought this would be covered by a separate case in `SynExpr`.
If that's correct, I'd suggest going through all of its cases and checking what is allowed in `nameof`.